### ISSUE TITLE
All Devices neighbors does't work any more in netdisco 2.39.21

### ISSUE THREAD

**Issue by def-bs** [netdisco/netdisco#393]
## Expected Behavior
 When I click on "All Devices" I expect to see all the devices in the network, since netdisco 2.39.21 such feature doesn't work any more

## Your Environment
Software	Version
App::Netdisco	2.39.21
SNMP::Info	3.54
DB Schema	52
PostgreSQL	9.6.7
Perl	                 5.24.1

Thank you,
Best Regards.

**Comment by zoeloe** [comment]
Confirmed. Not seeing the All devices map anymore.
App::Netdisco	2.39.21
SNMP::Info	3.54
DB Schema	52
PostgreSQL	9.5.12
Perl	5.22.1

my 2 cents

**Comment by zoeloe** [comment]
When trying the "All devices" map:
Just a single device?: NO device at all. Blank map.
 just a single device and its immediate neighbours?: NO device at all. Blank map.
No error message about "empty array"
"Only Neigbours" (default) map options works. Only "All Devices" is broken.

**Comment by ollyg** [comment]
@def-bs and @zoeloe I'm afraid we do need more detail before we can investigate, as I cannot reproduce your problem.

Can you look in the web logs as well? ~netdisco/logs/netdisco-web.log might have an SQL error. Also you can look in the Javascript console in your browser.

Many thanks

**Comment by zoeloe** [comment]
netdisco-web.log show the following error message when trying to view the "All Devices" netmap:
error request to GET /ajax/data/device/netmap crashed: Modification of a read-only value attempted at /home/netdisco/perl5/lib/perl5/App/Netdisco/Web/Plugin/Device/Neighbors.pm line 107

The javascript console show the following 3 errors when reloading the "All Devices" map en the 4th error is when clicking the "Redraw Map" button:

[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (bootstrap2-toggle.min.js.map, line 0)
[Error] Failed to load resource: the server responded with a status of 404 (Not Found) (underscore-min.map, line 0)
[Error] Failed to load resource: the server responded with a status of 500 (Internal Server Error) (netmap, line 0)
[Error] Failed to load resource: the server responded with a status of 500 (Internal Server Error) (netmap, line 0)

Safari java console messages

Thanks

**Comment by ollyg** [comment]
Thanks! I have applied a small fix which will be in the next point release.

**Comment by def-bs** [comment]
Hi Oliver,

thank you for you hard work.

I still have an issue with this, I am getting the error, even in the new release

[2616] 2018-04-19 06:00:39 error request to GET /ajax/data/device/netmap crashed: Modification of a read-only value attempted at /home/netdisco/perl5/lib/perl5/App/Netdisco/Web/Plugin/Device/Neighbors.pm line 107.

in the 2.39.22

**Comment by ollyg** [comment]
Thanks all. I have implemented an alternate fix which is in release 2.039023. Please let me know how you get on!

**Comment by def-bs** [comment]
Hi, 

I managed to install the 2.039023, but still getting the error.

**Comment by ollyg** [comment]
Hi @def-bs I cannot reproduce it then. Please would you open a new ticket (as this ticket is for something else) and also paste the error in the log and restart your web server to check?

**Comment by paecker** [comment]
Hi,
the fix worked for me

**Comment by ollyg** [comment]
Great! thanks all
